File: framework/languages/dotnet/samples/TicTacToe/Server/Play/Adapters/ZLink/Handlers/CreateGameHandler.cs
using TicTacToe.Server.Play.Application.GameCreation;
using TicTacToe.Shared.Contracts;
using Zlink.Framework.Contracts.Handlers;

namespace TicTacToe.Server.Play.Adapters.ZLink.Handlers;

[ZLinkHandlerGroup("play")]
sealed class CreateGameHandler(
    TicTacToeGameCreator games,
    ILogger<CreateGameHandler> logger)
{
    [ZLinkRequest]
    public async ValueTask<CreateGameRes> CreateAsync(
        CreateGameReq request,
        ZLinkRequestContext context,
        CancellationToken cancellationToken)
    {
        _ = context;
        logger.LogInformation(
            "api -> play: CreateGameReq received. game={GameName}",
            request.GameName);

        var created = await games.CreateAsync(request.GameName, cancellationToken);
        logger.LogInformation(
            "play: TicTacToeGame spot created. gameId={GameId}, endpoint={Endpoint}",
            created.GameId,
            created.PlayEndpoint);
        return created;
    }
}

File: framework/languages/node/samples/TicTacToe.Ts/Server/Play/Infrastructure/ZLink/Handlers/create-game-handler.ts
import { Inject } from '@nestjs/common';
import { TicTacToeGameCreator } from '../../../Application/GameCreation/tictactoe-game-creator';
import type {
  ZLinkRequestHandler
} from '@zlink-systems/framework';
import type { TicTacToeGameCreator as TicTacToeGameCreatorType } from '../../../Application/GameCreation/tictactoe-game-creator';
import type {
  CreateGameReq,
  CreateGameRes
} from '../../../../../Shared/Contracts/messages';

class CreateGameHandler implements ZLinkRequestHandler<CreateGameReq, CreateGameRes> {
  constructor(
    @Inject(TicTacToeGameCreator) private readonly games: TicTacToeGameCreatorType
  ) {}

  async handle(request: CreateGameReq): Promise<CreateGameRes> {
    return await this.games.create(request.gameName);
  }
}

export { CreateGameHandler };
